### ISSUE TITLE
HTTPs wizard

### ISSUE THREAD

**Issue by mitchhentges** [uxebu/james#43]
In the top bar, beside "URL Mappings", allow users to launch a modal that will allow them to setup the HTTPs requirements in-app. The wizard should:

1. Automatically generate the two SSL files. Progress bar?
2. Provide a convenient link to open the SSL certificate's location in a file manager
3. Explain that the certificate needs to be imported into browsers that will use the proxy
4. "Done" button at the bottom.

Alright, maybe the term "wizard" is a little too 90's to describe this feature, but `¯\_(ツ)_/¯`

**Comment by tomitm** [comment]
The 2010's term is "guide", I imagine. :wink:

Definitely a good idea. Thinking there should also be a message under/above the browser list warning that HTTPS proxying is disabled, with a `[Setup HTTPS]` button.

Not 100% sure about adding it to the top bar though, seems more suited for a settings window, y'know?

**Comment by nerdbeere** [comment]
Good idea, but this guide will differ from OS to OS.
In OSX for example I need to import the cert into my Keychain and then set the cert to "trust always"

**Comment by kokarn** [comment]
So, what is the way to do this at the moment?
Added certs to keychain, set as always trust, imported into firefox both after starting from james and before. 
James and both cert files in same folder and all that jazz as well.
Tried with chrome and safari with no luck :)

Maybe it's time for the "Wizard" ? ;)

**Comment by mitchhentges** [comment]
What part of it doesn't work? I'm assuming that you followed [the wiki?](https://github.com/james-proxy/james/wiki/Configuring-James-for-HTTPS).
All that the wizard will do is generate the `.key` and `.pem` files. After that, it's up to the user to apply the certificates to each browser/os, because that's a complicated problem to solve programmatically.

**Comment by kokarn** [comment]
Yeah that's true. Might be a lot of work for every browser + os combo.

Yeah I followed the wiki but I get different results in different browsers.

Firefox just doesn't proxy anything. Requests work but nothing appears in James. Same with Safari. Chrome was unable to connect at all.

**Comment by mitchhentges** [comment]
Can you create a new issue with screen shot and more details, per browser?

Did you launch the browser via James? What if you use your browser like
normal, but change your proxy settings manually?

It should be working, but if you generated your key and certificate OK,
then the wizard won't be able to help

**Comment by jariangibson** [comment]
Has anyone been able to get HTTPs/SSL working on Mac?

**Comment by mitchhentges** [comment]
Yes, I think so. I'll be able to test again in a few hours.

Anything in particular not working?

**Comment by jariangibson** [comment]
I create the certs, import in keychain & trust, & put in package contents where app executable is but no luck.

**Comment by mitchhentges** [comment]
The HTTPs/SSL not working was reported and fixed as per #221
